src/Microsoft.AspNet.Razor/RazorEditorParser.cs

```diff
@@ -12,45 +12,70 @@
-    /// the original TextChange that caused the reparse, to allow the editor to resolve the new tree against any changes made since
-    /// calling CheckForStructureChanges.
-    ///
-    /// If a call to CheckForStructureChanges occurs while a reparse is already in-progress, the reparse is cancelled IMMEDIATELY
-    /// and Rejected is returned without attempting to reparse.  This means that if a conusmer calls CheckForStructureChanges, which
-    /// returns Rejected, then calls it again before DocumentParseComplete is fired, it will only recieve one DocumentParseComplete
-    /// event, for the second change.
+    /// <list type="number">
+    /// <item><description>User edits document.</description></item>
+    /// <item><description>Editor builds a <see cref="TextChange"/> structure describing the edit and providing a
+    /// reference to the <em>updated</em> text buffer.</description></item>
+    /// <item><description>Editor calls <see cref="CheckForStructureChanges"/> passing in that change.
+    /// </description></item>
+    /// <item><description>Parser determines if the change can be simply applied to an existing parse tree node.
+    /// </description></item>
+    /// <list type="number">
+    /// <item><description>If it can, the Parser updates its parse tree and returns
+    /// <see cref="PartialParseResult.Accepted"/>.</description></item>
+    /// <item><description>If it cannot, the Parser starts a background parse task and returns
+    /// <see cref="PartialParseResult.Rejected"/>.</description></item>
+    /// </list>
+    /// </list>
+    /// NOTE: Additional flags can be applied to the <see cref="PartialParseResult"/>, see that <c>enum</c> for more
+    /// details. However, the <see cref="PartialParseResult.Accepted"/> or <see cref="PartialParseResult.Rejected"/>
+    /// flags will ALWAYS be present.
+    /// </para>
+    /// <para>
+    /// A change can only be incrementally parsed if a single, unique, <see cref="Span"/> (see
+    /// <see cref="Parser.SyntaxTree"/>) in the syntax tree can be identified as owning the entire change.
+    /// For example, if a change overlaps with multiple <see cref="Span"/>s, the change cannot be parsed incrementally
+    /// and a full reparse is necessary. A <see cref="Span"/> "owns" a change if the change occurs either a) entirely
+    /// within it's boundaries or b) it is a pure insertion (see <see cref="TextChange"/>) at the end of a
+    /// <see cref="Span"/> whose <see cref="Span.EditHandler"/> can accept the change (see
+    /// <see cref="SpanEditHandler.CanAcceptChange"/>).
+    /// </para>
+    /// <para>
+    /// Even if a single unique <see cref="Span"/> owner can be identified, it's possible the edit will cause the
+    /// <see cref="Span"/> to split or merge with other <see cref="Span"/>s, in which case, a full reparse is necessary
+    /// to identify the extent of the changes to the tree.
```

Comment:
Stating `it's possible the edit will cause the <see cref="Span"/> to split or merge with other <see cref="Span"/>s` makes it seem like those type of actions are possible via a partial parse. Partial parsing will at most add or delete a Symbol within an existing Span. It wont ever create a new Span or split an existing one. 

Part of the `SpanEditHandler.CanAcceptChange`s responsibility is to detect when a change has a greater impact than simply adding or deleting a symbol to prevent scenarios like this from ever happening.